simulator/main.rs

```diff
@@ -36,12 +37,32 @@ struct SimulatorOpts {
     page_size: usize,
 }
 
+trait Arbitrary {
+    fn arbitrary<R: Rng>(rng: &mut R) -> Self;
+}
+
+trait ArbitraryOf<T> {
```

Comment:
I feel like `ArbitraryFrom<T>` and `arbitrary_from()` are more accurate names for these?

take e.g.

```
// Arbitrary Value from Vec<&Value>
impl ArbitraryFrom<Vec<&Value>> for Value {
    fn arbitrary_from<R: Rng>(rng: &mut R, t: &Vec<&Value>) -> Self {
        if t.is_empty() {
            return Value::Null;
        }

        let index = rng.gen_range(0..t.len());
        t[index].clone()
    }
}
```

or

```
// Arbitrary Value from Columntype
impl ArbitraryFrom<ColumnType> for Value {
    fn arbitrary_from<R: Rng>(rng: &mut R, t: &ColumnType) -> Self {
        match t {
            ColumnType::Integer => Value::Integer(rng.gen_range(i64::MIN..i64::MAX)),
            ColumnType::Float => Value::Float(rng.gen_range(-1e10..1e10)),
            ColumnType::Text => Value::Text(gen_random_text(rng)),
            ColumnType::Blob => Value::Blob(gen_random_text(rng).as_bytes().to_vec()),
        }
    }
}
```

Reply:
Agreed, I had the same thought , but I had already written most of the code so just kept it as is at that point. I'll update them.